File: html/old/R2/SACO R2 2008 (Solutions)/heron.cpp
// SACO 2nd Round 2008
// Heron's Formula Solution - Keegan Carruthers-Smith

#include <cmath>
#include <iostream>

using namespace std;

int main() {
    int a,b,c;
    cout << "Enter a: ";
    cin >> a;
    cout << "Enter b: ";
    cin >> b;
    cout << "Enter c: ";
    cin >> c;
    double s = (a+b+c)/2.0;
    cout << sqrt(s * (s-a) * (s-b) * (s-c)) << endl;
    return 0;
}

File: html/old/R2/SACO R2 2008 (Solutions)/heron.pas
{
	SACO Round 2 200 8
	Solution to problem 2 - Heron's 
	by Ben Steenhuisen
} 
PROGRAM heron;

var 	a,b,c:integer;
	s,area:real;

BEGIN
	write("Enter a: ");
	read(a);
	write("Enter b: ");
	read(b);
	write("Enter c: ");
	read(c);
	s := (a+b+c+0.0)/(2.0);
	writeln(sqrt(s*(s-a)*(s-b)*(s-c)):5:9);
END.
